main.go

```diff
@@ -0,0 +1,114 @@
+package main
+
+import (
+	"github.com/tidwall/gjson"
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+	"time"
+)
+
+func validateCurrencyCode(CurrencyCode string) bool {
+	// description: validates currency code
+	result := false
+	// todo: work up a real test
+	if CurrencyCode == "is in list of currencies" {
+		result = true
+	} else {
+		result = false
+	}
+	return result
+}
+
+func getRate(CurrencyFrom string, CurrencyTo string) float64 {
+	httpposturl := fmt.Sprintf("https://query1.finance.yahoo.com/v7/finance/chart/%v%v%v", CurrencyFrom, CurrencyTo, "=x?corsDomain=finance.yahoo.com&range=1d&interval=1d")
+	// todo: get rid of bytes.NewBuffer yet set a request header
+	var jsonData = []byte(`{
+		"b": "l"
+	}`)
+
+	request, error := http.NewRequest("GET", httpposturl, bytes.NewBuffer(jsonData))
```

Comment:
What is `jsonData` for? 

Just `request, error := http.NewRequest("GET", httpposturl, nil)` I expect should be ok.

Reply:
`httpposturl` -> `httpGetUrl` as it is a GET request.

Reply:
nice find. 
will remove the protocol and method and use `url`!?

Reply:
just tested it, you were right:
nil is ok